dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -719,6 +727,176 @@ public Limits setUserLimits(@ApiParam(hidden = true) @Auth User authUser,
+                                      @ApiParam(value = "Git registry", required = true, allowableValues = "GITHUB_COM, GITLAB_COM, BITBUCKET_ORG") @QueryParam("gitRegistry") TokenType gitRegistry,
+                                      @ApiParam(value = "Git repository path", required = true) @QueryParam("repository") String repository) {
+        User foundUser = userDAO.findById(authUser.getId());
+
+        // Get all of the users source control tokens
+        List<Token> scTokens = workflowResource.checkOnBitbucketToken(foundUser)
+                .stream()
+                .filter(token -> Objects.equals(token.getTokenSource(), gitRegistry))
+                .collect(Collectors.toList());
+
+        // Add repository as workflow
+        if (scTokens.size() > 0) {
+            final Token gitToken = scTokens.get(0);
+            SourceCodeRepoInterface sourceCodeRepo = SourceCodeRepoFactory.createSourceCodeRepo(gitToken, client);
+            final String tokenSource = gitToken.getTokenSource().toString();
+
+            String gitUrl = "git@" + tokenSource + ":" + repository + ".git";
+            LOG.info("Adding " + gitUrl);
+
+            // Create a workflow stub object if necessary
+            final Optional<BioWorkflow> existingWorkflow = workflowDAO.findByPath(tokenSource + "/" + repository, false, BioWorkflow.class);
+            if (existingWorkflow.isEmpty()) {
+                final Workflow createdWorkflow = sourceCodeRepo.createStubBioworkflow(repository);
+                if (createdWorkflow != null) {
+                    final long workflowID = workflowDAO.create(createdWorkflow);
+                    final Workflow workflowFromDB = workflowDAO.findById(workflowID);
+                    workflowFromDB.getUsers().add(foundUser);
+                } else {
+                    String msg = "There was an error adding workflow with the path " + tokenSource + "/" + repository;
+                    LOG.error(msg);
+                    throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
+                }
+            } else {
+                String msg = "A workflow with path " + tokenSource + "/" + repository + " already exists.";
+                LOG.error(msg);
+                throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
+            }
+            Optional<BioWorkflow> finalWorkflow = workflowDAO.findByPath(tokenSource + "/" + repository, false, BioWorkflow.class);
+            bulkUpsertWorkflows(foundUser);
+            if (finalWorkflow.isPresent()) {
+                return finalWorkflow.get();
+            }
+        }
+        return null;
+    }
+
+    @DELETE
+    @Timed
+    @UnitOfWork
+    @Path("/workflows")
+    @Operation(operationId = "deleteWorkflow")
+    @ApiOperation(value = "Delete a workflow for a registry and repository path.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) })
```

Comment:
What prompts this from the workflow wizard?
This seems like it doesn't delete anything either (is this a stub?)
If so, needs javadoc to that end with something like a `//TODO`

Reply:
This is when you switch the toggle to off, though would only work for stub workflows (disable other workflows in the UI).

Line 819 deletes the workflow using
```
workflowDAO.delete(workflow);
```

Reply:
Ah ok, I don't know how I feel about this truly deleting things as opposed to archiving or making it inactive.

Reply:
This is only deleting stubs which I think is fine for now. Hopefully before this release is out we've come to a consensus on delete/archive.